app/iht/utils/pdf/XmlFoToPDF.scala

```diff
@@ -42,52 +47,79 @@ object XmlFoToPDF extends XmlFoToPDF
-      applicationDetails,
-      declaration,
-      declarationType,
-      applicationDetails.kickoutReason.isEmpty
-    )
+    val pdfoutStream = new ByteArrayOutputStream()
+
+    createPreSubmissionTransformer(regDetails, applicationDetails)
+      .transform(modelAsXMLStream, new SAXResult(fop(pdfoutStream).getDefaultHandler))
+    pdfoutStream.toByteArray
   }
 
-  private def createPreSubmissionPDF(template: StreamSource, modelSource: StreamSource, registrationDetails: RegistrationDetails,
-                                     applicationDetails: ApplicationDetails, declaration: Boolean, declarationType: String,
-                                     kickout: Boolean): Array[Byte] = {
-    val BASEURI = new File(".").toURI
-    val fopURIResolver = new FopURIResolver
-    val confBuilder = new FopConfParser(Play.classloader.getResourceAsStream(filePathForFOPConfig),
-      EnvironmentalProfileFactory.createRestrictedIO(BASEURI, fopURIResolver)).getFopFactoryBuilder
-    val fopFactory: FopFactory = confBuilder.build
+  def createPostSubmissionPDF(registrationDetails: RegistrationDetails, ihtReturn: IHTReturn): Array[Byte] = {
+    val modelAsXMLStream: StreamSource = new StreamSource(new ByteArrayInputStream(ModelToXMLSource.
+      getPostSubmissionDetailsXMLSource(registrationDetails, ihtReturn)))
 
-    val foUserAgent: FOUserAgent = fopFactory.newFOUserAgent
+    val pdfoutStream = new ByteArrayOutputStream()
+
+    createPostSubmissionTransformer(registrationDetails, ihtReturn)
+      .transform(modelAsXMLStream, new SAXResult(fop(pdfoutStream).getDefaultHandler))
+
+    pdfoutStream.toByteArray
+  }
+
+  def createClearancePDF(registrationDetails: RegistrationDetails, declarationDate: LocalDate): Array[Byte] = {
+    val modelAsXMLStream: StreamSource = new StreamSource(
+      new ByteArrayInputStream(ModelToXMLSource.getClearanceCertificateXMLSource(registrationDetails)))
 
     val pdfoutStream = new ByteArrayOutputStream()
+
+    createClearanceTransformer(registrationDetails, declarationDate)
+      .transform(modelAsXMLStream, new SAXResult(fop(pdfoutStream).getDefaultHandler))
+
+    pdfoutStream.write(pdfoutStream.toByteArray)
+    pdfoutStream.toByteArray
+  }
+
+  private def createClearanceTransformer(registrationDetails: RegistrationDetails,
+                                         declarationDate: LocalDate): Transformer = {
+    val template: StreamSource = new StreamSource(
+      Play.classloader.getResourceAsStream(filePathForClearanceXSL))
     val transformerFactory: TransformerFactory = TransformerFactory.newInstance
     transformerFactory.setURIResolver(new StylesheetResolver)
-
     val transformer: Transformer = transformerFactory.newTransformer(template)
+    setupTransformerEventHandling(transformer)
 
-    val fop: Fop = fopFactory.newFop(MimeConstants.MIME_PDF, foUserAgent, pdfoutStream)
-    val res = new SAXResult(fop.getDefaultHandler)
+    transformer.setParameter("pdfFormatter", PdfFormatter)
```

Comment:
can we possibly make some of the common parameters into a common method and set them that way